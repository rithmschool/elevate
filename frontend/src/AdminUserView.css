#InvoiceButton {
    float:right;
}
.AdminUserView {
  padding: 1vh 5vw;
  font-size: 1rem;
}