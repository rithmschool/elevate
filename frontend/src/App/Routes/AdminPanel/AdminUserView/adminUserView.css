.adminUserView_div {
  padding: 1vh 5vw;
  font-size: 1rem;
}
