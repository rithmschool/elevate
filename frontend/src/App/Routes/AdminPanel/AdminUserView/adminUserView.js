import React from "react";
import { withRouter, Redirect } from "react-router-dom";
import "./adminUserView.css";
import ElevateApi from "../../../../elevateApi";
import Spinner from "../../../Spinner/spinner";

class AdminUserView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {},
      redirect: false
    };
  }

  async componentDidMount() {
    try {
      const userId = this.props.match.params.userId;
      const user = await ElevateApi.getUser(userId);
      this.setState({ user });
    } catch (err) {
      console.log(err);
      this.setState({ redirect: true });
      return err;
    }
  }

  handleClickDeleteUser = async () => {
    await ElevateApi.deleteUser(this.props.match.params.userId);
    let users;

    try {
      users = await ElevateApi.getUsers();
    } catch (err) {
      console.log(err);
      return err;
    }
    this.props.updateUserState(users);
    this.props.history.push("/dashboard/admin/users");
  };

  render() {
    const {
      first_name,
      last_name,
      email,
      current_company,
      hire_date,
      needs,
      goals
    } = this.state.user;

    const { user, redirect } = this.state;

    if (!user) {
      return <Spinner />;
    }

    if (redirect) {
      return <Redirect to="/dashboard/admin/users" />;
    }

    return (
      <div className="adminUserView_div">
        <div>
          <h4>
            {first_name} {last_name}
          </h4>

          <table>
            <tbody>
              <tr>
                <td>
                  <b>Email:</b>
                </td>
                <td>{email}</td>
              </tr>
              <tr>
                <td>
                  <b>Company:</b>
                </td>
                <td>{current_company}</td>
              </tr>
              <tr>
                <td>
                  <b>Hire Date:</b>
                </td>
                <td>{hire_date && hire_date.slice(0, 10)}</td>
              </tr>
              <tr>
                <td>
                  <b>Needs:</b>
                </td>
                <td>{needs}</td>
              </tr>
              <tr>
                <td>
                  <b>Goals:</b>
                </td>
                <td>{goals}</td>
              </tr>
              <tr>
                <td>
                  <b>Questions:</b>
                </td>
                {/* Add user questions here */}
              </tr>
            </tbody>
          </table>
        </div>

        <button
          onClick={e => {
            if (window.confirm("Are you sure you want to delete this user?")) {
              this.handleClickDeleteUser(e);
            }
          }}
          id="delete-click"
        >
          Delete
        </button>
      </div>
    );
  }
}

export default withRouter(AdminUserView);
