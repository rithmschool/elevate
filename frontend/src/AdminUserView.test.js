import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import AdminUserView from './adminUserView';

describe('AdminUserView', function() {
  let wrapper;
  const testData = { 
    first_name: 'Han', 
    last_name: 'Solo', 
    email: 'han@solo.com', 
    current_company: 'SpaceX', 
    hire_date: '2018-09-01', 
    needs: 'Negotiate a raise during annual review', 
    goals: 'Increase salary by 15k'
  }

  beforeEach(() => {
    wrapper = mount(<AdminUserView user={testData}/>);
  });

  it('renders without crashing', function () {
    // Mounted in beforeEach above
  });
  
  it('matches snapshot', function () {
    const serialized = toJson(wrapper);

    expect(serialized).toMatchSnapshot();
  });

  it('has a user prop', function () {
    expect(wrapper.prop('user')).toEqual(testData);
  });

  it('has div with AdminUserView class', function () {
    expect(wrapper.find('div.AdminUserView')).toHaveLength(1);
  });

  it('has user info fields', function () {
    expect(wrapper.html()).toContain('Email');
    expect(wrapper.html()).toContain('Company');
    expect(wrapper.html()).toContain('Hire Date');
    expect(wrapper.html()).toContain('Needs');
    expect(wrapper.html()).toContain('Goals');
    expect(wrapper.html()).toContain('Questions');
  });
});