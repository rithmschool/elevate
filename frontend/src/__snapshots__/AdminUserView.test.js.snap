// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUserView matches snapshot 1`] = `
<AdminUserView
  user={
    Object {
      "current_company": "SpaceX",
      "email": "han@solo.com",
      "first_name": "Han",
      "goals": "Increase salary by 15k",
      "hire_date": "2018-09-01",
      "last_name": "Solo",
      "needs": "Negotiate a raise during annual review",
    }
  }
>
  <div
    className="AdminUserView"
  >
    <div>
      <h4>
        Han
         
        Solo
      </h4>
      <table>
        <tbody>
          <tr>
            <td>
              <b>
                Email:
              </b>
            </td>
            <td>
              han@solo.com
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Company:
              </b>
            </td>
            <td>
              SpaceX
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Hire Date:
              </b>
            </td>
            <td>
              2018-09-01
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Needs:
              </b>
            </td>
            <td>
              Negotiate a raise during annual review
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Goals:
              </b>
            </td>
            <td>
              Increase salary by 15k
            </td>
          </tr>
          <tr>
            <td>
              <b>
                Questions:
              </b>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</AdminUserView>
`;
